x/distribution/client/cli/tx.go

```diff
@@ -40,6 +45,28 @@ func GetTxCmd(storeKey string, cdc *amino.Codec) *cobra.Command {
 	return distTxCmd
 }
 
+func splitGenerateOrBroadcast(cliCtx context.CLIContext, txBldr authtxb.TxBuilder, msgs []sdk.Msg) error {
+	chunkSize := viper.GetInt(flagMaxMessagesPerTx)
+	totalMessages := len(msgs)
+	if chunkSize == 0 {
+		chunkSize = totalMessages
+	}
+
+	for i := 0; i < len(msgs); i += chunkSize {
+		sliceEnd := i + chunkSize
+		if sliceEnd > totalMessages {
+			sliceEnd = totalMessages
+		}
+		msgChunk := msgs[i:sliceEnd]
+		err := utils.GenerateOrBroadcastMsgs(cliCtx, txBldr, msgChunk)
```

Comment:
Let's do this inline:
```suggestion
		if err := utils.GenerateOrBroadcastMsgs(cliCtx, txBldr, msgChunk); err != nil {
			return err
		}
```